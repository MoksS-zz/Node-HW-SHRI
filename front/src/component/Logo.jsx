import React from "react";
import Button from "./Button";
import Text from "./Text";

const Logo = () => (
  <>
    <div className="Logo">
      <img className="Logo_Img" src="/icon/logo 3.svg" alt="LOGO" />
      <p className="Logo_Text">Configure repository connection and synchronization settings</p>
      <Button style={{ color: "action", padding: "action" }}>
        <Text style={{ size: "m", lineHeight: "xxxxl", weight: "small", color: "default" }}>Open settings</Text>
      </Button>
    </div>
  </>
)


export default Logo;