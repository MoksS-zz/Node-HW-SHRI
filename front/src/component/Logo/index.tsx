import React, { FC } from "react";
import { Link } from "react-router-dom";
import Button from "./../Button";
import Text from "./../Text";
import "./Logo.scss";
import logoImg from "../../icon/logo.svg"

const Logo: FC = () => {
  const { lang } = window;

  return (
    <>
      <div className="Logo">
        <img className="Logo_Img" src={logoImg} alt="LOGO" />
        <p className="Logo_Text">
          {lang.Logo.Text}
        </p>
        <Link to="/settings">
          <Button style={{ color: "action", padding: "action" }}>
            <Text style={{ size: "m", lineHeight: "xxxxl", weight: "small", color: "default" }}>
              {lang.Logo.Button.OpenSettings}
            </Text>
          </Button>
        </Link>
      </div>
    </>
  )
}

export default Logo;