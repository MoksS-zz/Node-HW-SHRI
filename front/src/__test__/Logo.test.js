import React from "react";
import renderer from "react-test-renderer";
import Logo  from "../component/Logo";

jest.mock('react-router-dom', () => ({
  Link: (props) => <a href={props.to}>{props.children}</a>
}))



it("renders Logo correctly", () => {
  const logo = renderer
    .create(<Logo/>)
    .toJSON();

  expect(logo).toMatchSnapshot();
});

