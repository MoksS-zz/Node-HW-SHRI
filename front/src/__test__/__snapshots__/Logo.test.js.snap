// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Logo correctly 1`] = `
<div
  className="Logo"
>
  <img
    alt="LOGO"
    className="Logo_Img"
    src="logo.svg"
  />
  <p
    className="Logo_Text"
  >
    Configure repository connection and synchronization settings
  </p>
  <a
    href="/settings"
  >
    <button
      className="Button Button__color-action Button__padding-action"
    >
      <span
        className="Text Text__size-m Text__lineHeight-xxxxl Text__weight-small Text__color-default"
      >
        Open settings
      </span>
    </button>
  </a>
</div>
`;
